avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,132 @@
+  }
+}
+
+// Diagnostics
+// ===========
+module diagnosticDependencies '../../../../../../utilities/e2e-template-assets/templates/diagnostic.dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}01'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-${namePrefix}-evh-${serviceShort}'
+    eventHubNamespaceName: 'dep-${namePrefix}-evhns-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    publicNetworkAccess: 'Enabled'
+    fhirservices: [
+      {
+        name: '${namePrefix}-az-fhir-x-001'
+        location: resourceLocation
+        kind: 'fhir-R4'
+        workspaceName: '${namePrefix}${serviceShort}001'
+        corsOrigins: ['*']
+        corsHeaders: ['*']
+        corsMethods: ['GET']
+        corsMaxAge: 600
+        corsAllowCredentials: false
+        diagnosticSettings: [
+          {
+            eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+            eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+            storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+            workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+          }
+        ]
+        publicNetworkAccess: 'Enabled'
+        resourceVersionPolicy: 'versioned'
+        smartProxyEnabled: false
+        managedIdentities: {
+          systemAssigned: false
+          userAssignedResourceIds: [
+            nestedDependencies.outputs.managedIdentityResourceId
+          ]
+        }
+        importEnabled: false
+        initialImportMode: false
+      }
+    ]
+    dicomservices: [
+      {
+        name: '${namePrefix}-az-dicom-x-001'
+        location: resourceLocation
+        workspaceName: '${namePrefix}${serviceShort}001'
+        corsOrigins: ['*']
+        corsHeaders: ['*']
+        corsMethods: ['GET']
+        corsMaxAge: 600
+        corsAllowCredentials: false
+        diagnosticSettings: [
+          {
+            eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+            eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+            storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+            workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+          }
+        ]
+        publicNetworkAccess: 'Enabled'
+        managedIdentities: {
```

Comment:
Are all these parameters required for WAF-alignment?

Reply:
removed the child module tests